### DISCUSSION TITLE
useTreeData runtime error  Cannot read properties of undefined when reading map.

### DISCUSSION THREAD

**Discussion by kyosimba** [adobe/react-spectrum#2546] [Q&A]
I tried to run the example [https://react-spectrum.adobe.com/react-stately/useTreeData.html#example](here) . It seems that when mapping over `initialItems` I get cannot read properties of undefined. I am using `@react-stately/data ^3.4.2` what am I missing.
```js
let  tree = useTreeData({
    initialItems: [
      {
        name: "People",
        items: [{ name: "David" }, { name: "Sam" }, { name: "Jane" }],
      },
      {
        name: "Animals",
        items: [{ name: "Aardvark" }, { name: "Kangaroo" }, { name: "Snake" }],
      },
    ],
    initialSelectedKeys: ["Sam", "Kangaroo"],
    getKey: (item) => item.name,
    getChildren: (item) => item.items,
  });
```

**Comment by reidbarber** [comment]
This looks like a bug in useTreeData where it is trying to map over undefined children of children (i.e. `initialItems[0].items[0].items`)

A temporary fix on your end would be to use:
```js
getChildren: (item) => item.items || [],
```

**Reply by kyosimba** [reply]
thanks that helped

**Reply by reidbarber** [reply]
This should be fixed via https://github.com/adobe/react-spectrum/pull/2549 in nightly builds as of tonight, so the workaround is no longer needed.
